<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<arg name="description" default="cbf_codegen_test"/>

	<arg name="margin" default="30.0"/>
	<arg name="use_margin" default="false"/>
	<arg name="hwil" default="true"/>
	<arg name="readonly" default="false"/>
	<arg name="vsl_test" default="true"/>

<!-- 	<arg name="timegap" default="2.0"/>
	<arg name="k_timegap" default="0.1"/>
	<arg name="s_min" default="15.0"/>
	<arg name="k1" default="0.5"/>
	<arg name="k2" default="0.5"/>

	<param name="/margin" value="$(arg margin)"/>
	<param name="/use_margin" value="$(arg use_margin)"/>
	<param name="/hwil" value="$(arg hwil)"/>
	<param name="/readonly" value="$(arg readonly)"/>

	<param name="/timegap" value="$(arg timegap)"/>
	<param name="/k_timegap" value="$(arg k_timegap)"/>
	<param name="/s_min" value="$(arg s_min)"/>
	<param name="/k1" value="$(arg k1)"/>
	<param name="/k2" value="$(arg k2)"/> -->

	<!-- <node pkg="can_to_ros" type ="subs_fs" name="subs_fs" output="screen" if="$(arg hwil)"/>  -->
	<node pkg="can_to_ros" type ="subs" name="subs" output="screen" if="$(arg hwil)">
	</node>

	<node pkg="can_to_ros" type="lead_info" name="lead_info" output="screen" if="$(arg hwil)"/>

	<node pkg="can_to_ros" type="simple_mini_car_from_lead_distance" name="simple_mini_car_from_lead_distance" output="screen" if="$(arg hwil)"/>

	<!--node pkg="can_to_ros" type="cruise_start_safety_check" name="cruise_start_safety_check" output="screen" if="$(arg hwil)"/-->


	<node pkg="margin" type="margin_node" name="margin_node" output="screen" if="$(arg use_margin)">
	</node>

	<node pkg="setpointreader" type="setpointreader" name="setpointreader" output="screen" required="true" >
		<!-- <remap from="/v_ref" to="/cmd_vel"/> -->
		<!-- pcm_cruise_2 to msg_467 for the setpointreader to work -->
			<remap from="/msg_467" to="/pcm_cruise_2"/>
			<remap if="$(arg vsl_hwil_test)" from="/v_ref" to="/cmd_vel"/>
	</node>

	<node pkg="velocity_controller" type="velocity_controller_node" name="velocity_controller_node" output="screen">
		<remap from="/vehicle/vel" to="/vel"/>
		<remap from="/cmd_accel" to="/cmd_accel_pre"/>
	</node>

	<node pkg="cbf" type="cbf_node" name="cbf_node" if="$(eval not readonly and hwil)">
	</node>

	<node pkg="cbf" type="cbf_node" name="cbf_node_readonly" if="$(eval not readonly and hwil)">
		<remap from="/cmd_accel" to="/cmd_accel_null"/>
	</node>


	<node pkg="can_to_ros" type ="rosbag_record.sh" name="bashscript2" output="screen" args="$(arg description) $(arg hwil)" />

	<!-- <node pkg="transfer_pkg" type="saveparam.py" name="saveparam" output="screen" args="$(arg description) $(arg hwil)"> -->
	<!-- this is deprecated until updated -->
	<!-- </node> -->


</launch>
